learning_journal/views.py

```diff
@@ -1,21 +1,77 @@
 from pyramid.response import Response
 from pyramid.view import view_config
+from pyramid.httpexceptions import HTTPFound
 
-from sqlalchemy.exc import DBAPIError
+from sqlalchemy.exc import DBAPIError, SQLAlchemyError
+from sqlalchemy import desc
 
 from .models import (
     DBSession,
     Entry,
     )
 
+from wtforms import Form, StringField, TextAreaField, validators
+import markdown
 
-@view_config(route_name='home', renderer='templates/mytemplate.pt')
-def my_view(request):
-    try:
-        one = DBSession.query(Entry).filter(Entry.name == 'one').first()
-    except DBAPIError:
-        return Response(conn_err_msg, content_type='text/plain', status_int=500)
-    return {'one': one, 'project': 'learning-journal'}
+class EntryForm(Form):
+    title = StringField(u'Title', [validators.required(),
+                        validators.length(max=128)])
+    text = TextAreaField(u'Entry', [validators.required()])
+
+
+@view_config(route_name='list', renderer='templates/list_template.jinja2')
+def list_view(request):
+    entries = DBSession.query(Entry).order_by(desc(Entry.created))
+    return {'entries': entries, 'title': 'Learning Journal'}
+
+
+@view_config(route_name='detail', renderer='templates/detail_template.jinja2')
+def detail_view(request):
+    entry = DBSession.query(Entry).filter(
+        Entry.id == request.matchdict['id']).first()
+    title = "Learning Journal Entry {}".format(request.matchdict['id'])
+    return {'entry': entry, 'entry_text': markdown.markdown(entry.text), 'title': title}
+
+
+@view_config(route_name='add_entry',
+             renderer='templates/add_entry_template.jinja2')
+def add_entry_view(request):
+    entry_form = EntryForm(request.POST)
+    if request.method == 'POST' and entry_form.validate():
+        try:
+            new_entry = Entry(title=entry_form.title.data,
+                              text=entry_form.text.data)
+            DBSession.add(new_entry)
+        except SQLAlchemyError:
+            return {'title': 'Add Entry', 'form': entry_form,
+                    'error': 'Title Already Used'}
```

Comment:
SQLAlchemyError seems like it might be raised for reasons other than a duplicate entry title.  I suggest you find a more specific error to catch if you plan on returning an error message this specific.